crates/polars/tests/it/lazy/predicate_queries.rs

```diff
@@ -269,3 +269,33 @@ fn test_predicate_pushdown_block_8847() -> PolarsResult<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_push_join_col_predicates_to_both_sides_7247() -> PolarsResult<()> {
+    let df1 = df! {
+        "a" => ["a1", "a2"],
+        "b" => ["b1", "b2"],
+    }?;
+    let df2 = df! {
+        "a" => ["a1", "a1", "a2"],
+        "b2" => ["b1", "b1", "b2"],
+        "c" => ["a1", "c", "a2"]
+    }?;
+    let df = df1.lazy().join(
+        df2.lazy(),
+        [col("a"), col("b")],
+        [col("a"), col("b2")],
+        JoinArgs::new(JoinType::Inner),
+    );
+    let out = df
+        .filter(col("a").eq(lit("a1")))
+        .filter(col("a").eq(col("c")))
+        .collect()?;
+    let expected = df![
+        "a" => ["a1"],
+        "b" => ["b1"],
+        "c" => ["a1"],
+    ]?;
+    assert_eq!(out, expected);
```

Comment:
This tests if the result is correct, but not if the optimization is applied to the right side. Can you test that?

Reply:
You you can iter the logical plans and assert all scan nodes have a predicate.

Maybe you can add an argument to this function: https://github.com/pola-rs/polars/blob/696a1f1ec784cd5e90903259c58e9f0c30f7d574/crates/polars-lazy/src/tests/optimization_checks.rs#L22

Where the argument influence if we do an `all` or an `any` check.

Reply:
Or maybe check if all tests still succeed if you change it to `all` and go for that.